### PR TITLE
Mkdocs gallery and `sys.path`

### PR DESCRIPTION
**Pull request by Louis-Pujol**
Hello,

I have a little issue with `sys.path` and `mkdocs-gallery`. I realized that changes to sys.path were not persistent across code blocks. Here is a simple illustration script :

```python
"""
PYTHONPATH
==========
"""

# %%
# Copy python path

import sys
original_path = sys.path.copy()


# %%
# Append path and print new path directories

sys.path.append("/home/mypath")
new_path = sys.path.copy()
for p in new_path:
    if p not in original_path:
        print(p)


# %%
# Print new path directories

path2 = sys.path.copy()
for p in path2:
    if p not in original_path:
        print(p)
```

And the resulting example:
Capture d’écran du 2023-11-14 15-13-53

`sys.path` Were reinitialized between block 2 and block 3 as `/home/mypath` is no longer inside `sys.path`.This behavior comes from this function, which is called between blocks and reset sys.path: 

https://github.com/smarie/mkdocs-gallery/blob/d4c2d7fd7991a590abc2c947a61334b3a56ab9f7/src/mkdocs_gallery/gen_single.py#L731-#L735

This caused me some trouble because I use some libraries that add cache directories to `sys.path`. If I understand correctly the purpose of `_reset_cwd_path`, it is meant to remove cwd from `sys.path`, but it actually removes all new additions.

In this PR, I propose to modify `_reset_cwd_path` in order to remove only the last cwd and do not erase other modifications to `sys.path`. With this new version, my example works as I expected:

Capture d’écran du 2023-11-14 15-25-39

Let me know what are your thoughts about this changes.
Many thanks,
Louis

### CODE REVIEW

File: src/mkdocs_gallery/gen_single.py [hunk 1]
```diff
@@ -728,10 +728,11 @@ def _get_code_output(is_last_expr, script: GalleryScript, logging_tee, images_md
     return code_output
 
 
-def _reset_cwd_syspath(cwd, sys_path):
+def _reset_cwd_syspath(cwd, new_path):
```
Comment (smarie) [inline src/mkdocs_gallery/gen_single.py:731]:
```suggestion
def _reset_cwd_syspath(cwd, tmp_path_to_remove):
```

File: src/mkdocs_gallery/gen_single.py [hunk 2]
```diff
@@ -728,10 +728,11 @@ def _get_code_output(is_last_expr, script: GalleryScript, logging_tee, images_md
...
     """Reset cwd and sys.path."""
```
Comment (smarie) [inline src/mkdocs_gallery/gen_single.py:732]: Please replace this with `"""Reset current working directory to `cwd` and remove `path_to_remove` from `sys.path`."""`

File: src/mkdocs_gallery/gen_single.py [hunk 3]
```diff
@@ -773,8 +774,8 @@ def execute_code_block(compiler, block, script: GalleryScript):
     # created by the example get created in this directory
     os.chdir(src_file.parent)
 
-    sys_path = copy.deepcopy(sys.path)
-    sys.path.append(os.getcwd())
+    new_path = os.getcwd()
+    sys.path.append(new_path)
```
Comment (smarie) [inline src/mkdocs_gallery/gen_single.py:778]:
```suggestion
    # Add the example dir to the path temporarily (will be removed after execution)
    new_path = os.getcwd()
    sys.path.append(new_path)
```

**Comment by smarie** [comment]
Thanks @Louis-Pujol for this proposal ! 
I made a few readability comments (although the original sphinx-gallery code is not the best in readability, but we can at least marginally add comments)

Could you please also add a changelog entry for your mod ? Thanks !

**Comment by Louis-Pujol** [comment]
Thanks, I committed your suggestions.

I'm not sure about what I need to do for adding a changelog entry. Do I need to create a new release on my repository ? And if yes, which tag do I need to choose ? Sorry if my questions are naive, I'm not very familiar with those things

**Comment by smarie** [comment]
Thanks @Louis-Pujol . No, simply modify the file `docs/changelog.md` : add a bullet point in the first section (`0.7.9 - (In Progress)`) stating your contribution. You can follow the example that is already there

**Comment by Louis-Pujol** [comment]
Thanks, I just did it.

**Comment by Louis-Pujol** [comment]
Thank you for pointing this out. I did the resetting of `sys.path` with a `copy.deepcopy()` + merge the last changes.

**Comment by smarie** [comment]
Perfect @Louis-Pujol  ! Thanks a lot. Will this this now. Two releases on the same day :D

**Pull request merged**